verification/cbmc/proofs/aws_byte_cursor_read_and_fill_buffer/aws_byte_cursor_read_and_fill_buffer_harness.c

```diff
@@ -43,7 +43,9 @@ void aws_byte_cursor_read_and_fill_buffer_harness() {
             assert_byte_from_buffer_matches(cur.ptr, &old_byte_from_cur);
         }
     } else {
-        assert(cur.ptr == old_cur.ptr + old_buf.capacity);
+        if (old_cur.ptr) {
+            assert(cur.ptr == old_cur.ptr + old_buf.capacity);
+        }
```

Comment:
Use `AWS_IMPLIES`? See first example.

Reply:
Hmm, it seems to me like we are losing something here after this change.
Now we don't seem to check anything if `old_cur.ptr` is `NULL`.

How about:
```c
if (old_cur.ptr) {
    assert(cur.ptr == old_cur.ptr + old_buf.capacity);
} else {
    assert(cur.ptr == NULL);
}
```